pyrefinebio/util.py

```diff
@@ -14,6 +17,20 @@ def parse_date(date):
         return parsed
 
 
+def expand_path(path, filename):
+    user_path = os.path.expanduser(path)
+    full_path = os.path.abspath(user_path)
+
+    if os.path.isdir(full_path):
+        full_path = os.path.join(full_path, filename)
+
+    return full_path
+
+
+def extract(path):
+    shutil.unpack_archive(path)
```

Comment:
I think it'd be better if you didn't rename this function. Reading the other files, it's less clear what it's doing until you come here. Also calling an `extract` function from the `extract` methods is slightly confusing like "wait, is this recursive? oh no I'm in a class".